tools/psa/build_tfm.py

```diff
@@ -18,22 +18,24 @@
 """
 
 import os
-from os.path import join, abspath, dirname, isdir
+from os.path import join, abspath, dirname, isdir, relpath
+import argparse
 import sys
 import shutil
 import subprocess
 import logging
 
-logger = logging.getLogger('TF-M-Builder')
+ROOT = abspath(join(dirname(__file__), os.pardir, os.pardir))
+sys.path.insert(0, ROOT)
+from tools.targets import Target, TARGET_MAP, TARGET_NAMES
+
 logging.basicConfig(level=logging.INFO,
                     format='[%(name)s] %(asctime)s: %(message)s.',
                     datefmt='%H:%M:%S')
+logger = logging.getLogger('TF-M-Builder')
 
-ROOT = abspath(join(dirname(__file__), os.pardir, os.pardir))
-sys.path.insert(0, ROOT)
-
-TF_M_BUILD_DIR = join(ROOT, os.pardir, 'tfm_build_dir')
-VERSION_FILE_PATH = join(ROOT, 'features/FEATURE_PSA/FEATURE_TFM')
+TF_M_BUILD_DIR = abspath(join(ROOT, os.pardir, 'tfm_build_dir'))
```

Comment:
Do we want to build into `BUILD` folder?

Reply:
@Patater Unfortunately it is not possible to filter out directories with Mbed OS build system. If we use `BUILD` folder then if one tries to build non-secure side (Mbed OS) after build secure side (TF-M) then Mbed OS build system will try to build secure side and fail.

Reply:
Please use a temp folder, via Python's tempfile API.

Reply:
@Patater Added new command line option `--develop`. By default a temporary folder is used as tf-m build folder. But when `--develop` command line option is used then a new folder (tfm_build_dir) under the parent folder of Mbed OS is used as tf-m build folder